src/client/__tests__/get-data.spec.js

```diff
@@ -0,0 +1,100 @@
+import getData from '../get-data';
+
+describe( 'getData', () => {
+	it( 'should return null if clientState does not exist.', () => {
+		const data = getData( null )( [ 'things', 1 ] );
+		expect( data ).toBeNull();
+	} );
+	it( 'should return null if endpoint does not exist.', () => {
+		const clientState = {
+			endpoints: {
+				things: {
+				},
+			},
+		};
+		const data = getData( clientState )( [ 'things', 1 ] );
+		expect( data ).toBeNull();
+	} );
+	it( 'should return null if data is empty.', () => {
+		const clientState = {
+			endpoints: {
+				things: {
+					endpoints: {
+						1: {},
+					},
+				},
+			},
+		};
+		const data = getData( clientState )( [ 'things', 1 ] );
+		expect( data ).toBeNull();
```

Comment:
Why wouldn't this return an empty object?

Reply:
Because an empty object may be valid API data. Null indicates the absence of data.